tests/Unit/Pypp/PyppFundamentals.hpp

```diff
@@ -406,4 +412,128 @@ struct FromPyObject<DataVector, std::nullptr_t> {
     return t;
   }
 };
+
+template <typename T>
+struct FromPyObject<Scalar<T>> {
+  static Scalar<T> convert(PyObject* p) {
+    return Scalar<T>{from_py_object<T>(p)};
+  }
+};
+
+template <typename T, Requires<tt::is_a_v<Tensor, T>> = nullptr>
```

Comment:
why cant you use the one inside tensor structure?

Reply:
I didn't see it in Structure? There's a tensor_index from collapsed function, and a collapsed_to_storage which takes into account the symmetry of the tensor which we don't want to do here.

Reply:
can we use IndexIterator?

Reply:
I think I can use IndexIterator but I have to change two things about index: 
- I've to be able to get a std::array<size_t> from an Index
- I've to allow the Dim parameter of index to take values up to 4 (or more if we have larger rank tensors).